src/annotator/profile.tsx

```diff
@@ -0,0 +1,29 @@
+import { render } from 'preact';
+
+import type { Destroyable } from '../types/annotator';
+
+import type { ProfileConfig } from './components/ProfileModal';
+import { ProfileModal } from './components/ProfileModal';
+import type { EventBus } from './util/emitter';
+import { createShadowRoot } from './util/shadow-root';
+
+export class Profile implements Destroyable {
+  private _outerContainer: HTMLElement;
+  public shadowRoot: ShadowRoot;
+
+  constructor(element: HTMLElement, eventBus: EventBus, config: ProfileConfig) {
+    this._outerContainer = document.createElement('hypothesis-profile');
+    element.appendChild(this._outerContainer);
+    this.shadowRoot = createShadowRoot(this._outerContainer);
+
+    render(
+      <ProfileModal eventBus={eventBus} config={config} />,
+      this.shadowRoot
+    );
+  }
+
+  destroy(): void {
+    render(null, this.shadowRoot);
+    this._outerContainer.remove();
+  }
+}
```

Comment:
This is mostly duplicating what's in `Notebook` class.

Maybe we could also unify it somehow, and then do something in the lines of:

```ts
const notebook = new WhateverDescriptiveAndClearName(...);
const profile = new WhateverDescriptiveAndClearName(...);
```

Instead of

```ts
const notebook = new Notebook(...);
const profile = new Profile(...);
```

But I have some doubts. It make make it less clear.

Reply:
Agree